### PR TITLE
buildkite: support version 21 for opentelemetry

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Use java 21.
Fixes https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/11267

## Test

image

## Checklist

- [ ] This is an enhancement of existing features, or a new feature in existing plugins
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] I have made corresponding changes to the documentation
- [ ] This is a bugfix
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that would fail without this fix
- [ ] This is a new plugin
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] My code follows the [style guidelines of this project](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#java-language-formatting-guidelines)
  - [ ] I have made corresponding changes to the documentation
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#testing) pass locally with my changes
  - [ ] I have updated [supported-technologies.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/docs/supported-technologies.asciidoc)
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] Added an instrumentation plugin? Describe how you made sure that old, non-supported versions are not instrumented by accident.
- [ ] This is something else
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)

### CODE REVIEW

File: .buildkite/hooks/pre-command
Comment (v1v) [inline .buildkite/hooks/pre-command:18]: Since I changed when to install things, I ran the snapshot workflow manually:
- https://buildkite.com/elastic/apm-agent-java-snapshot/builds/397

In addition, I'm also validating these changes in the Buildkite pipeline:
- https://buildkite.com/elastic/apm-agent-java-opentelemetry-benchmark/builds/46

**Review by amannocci — APPROVED** [review]

**Review by JonasKunz — APPROVED** [review]

**Comment by v1v** [comment]
@jackshirazi , IIRC, you said we cannot use 21 at the moment; if that's the case, this should wait for a final decision.

File: .buildkite/hooks/pre-command
Comment (JonasKunz) [reply .buildkite/hooks/pre-command:18]: Was the snapshot actually run on this branch?
According to the logs it still ran on Java 17 and I can't see how this would be the case with this PR:
```
Java version: 17.0.2, vendor: Oracle Corporation, runtime: /usr/java/openjdk17
```

We will need to stick to Java 17 for building the agent for Java 7 support.

Comment (v1v) [reply .buildkite/hooks/pre-command:18]: This change won't affect snapshot/releases Buildkite pipelines but the opentelemetry benchmark only:
- https://buildkite.com/elastic/apm-agent-java-opentelemetry-benchmark/builds?branch=fix%2Fjava-21-otel

image

the it failed with

```

> Task :smoke-tests-otel-starter:spring-smoke-testing:bootJar FAILED
--
  |  
  | FAILURE: Build failed with an exception.
  |  
  | * What went wrong:
  | Execution failed for task ':smoke-tests-otel-starter:spring-smoke-testing:bootJar'.
  | > Error while evaluating property 'mainClass' of task ':smoke-tests-otel-starter:spring-smoke-testing:bootJar'.
  | > Failed to calculate the value of task ':smoke-tests-otel-starter:spring-smoke-testing:bootJar' property 'mainClass'.
  | > Main class name has not been configured and it could not be resolved


```

when running https://github.com/elastic/apm-agent-java/blob/ea685113257279c820dc4b97d288486fb7cbe76d/.buildkite/scripts/opentelemetry-benchmark.sh#L39

Comment (JonasKunz) [reply .buildkite/hooks/pre-command:18]: Ok then I don't see a reason why this shouldn't work, if this only affects the benchmark. @jackshirazi ?

Comment (jackshirazi) [reply .buildkite/hooks/pre-command:18]: yes, let's go ahead

**Review by jackshirazi — APPROVED** [review]

**Pull request merged**
